runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -0,0 +1,141 @@
+import org.corfudb.runtime.exceptions.WorkflowResultUnknownException;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.util.Sleep;
+
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+/**
+ * An abstract class that defines a generec workflow request structure.
+ * <p>
+ * Created by Maithem on 1/19/18.
+ */
+@Slf4j
+public abstract class WorkflowRequest {
+
+    protected int retry;
+
+    protected Duration timeout;
+
+    protected Duration pollPeriod;
+
+    protected CorfuRuntime runtime;
+
+    @NonNull
+    protected String nodeForWorkflow;
+
+    /**
+     * Send a workflow request
+     *
+     * @param layout current layout
+     * @return a uuid that corresponds to the created workflow
+     */
+    protected abstract UUID sendRequest(Layout layout);
+
+    /**
+     * Select an orchestrator and return a client. Orchestrator's that
+     * are on unresponsive nodes and the affected endpoint by the workflow
+     * will not be selected.
+     *
+     * @param layout the layout that is used to select an orchestrator
+     * @return a management client that is connected to the selected
+     * orchestrator
+     */
+    protected ManagementClient getOrchestrator(Layout layout) {
+        List<String> activeLayoutServers = layout.getLayoutServers().stream()
+                .filter(s -> !layout.getUnresponsiveServers().contains(s)
+                        && !s.equals(nodeForWorkflow))
+                .collect(Collectors.toList());
+
+        if (activeLayoutServers.isEmpty()) {
+            throw new WorkflowException("getOrchestrator: no available orchestrators " + layout);
+        }
+
+        return runtime.getRouter(activeLayoutServers.get(0)).getClient(ManagementClient.class);
+    }
+
+    /**
+     * Infer the completion of the request by inspecting the layout
+     *
+     * @param layout the layout to inspect
+     * @return true if the operation is reflected in the layout, otherwise
+     * return false
+     */
+    protected abstract boolean verifyRequest(Layout layout);
+
+    /**
+     * @param workflow   the workflow id to poll
+     * @param client     a client that is connected to the orchestrator that is
+     *                   running the workflow
+     * @param timeout    the total time to wait for the workflow to complete
+     * @param pollPeriod the poll period to query the completion of the workflow
+     * @throws NetworkException if the client disconnects
+     * @throws TimeoutException if the workflow doesn't complete withint the timout
+     *                          period
+     */
+    private void waitForWorkflow(@Nonnull UUID workflow, @Nonnull ManagementClient client,
+                                 @Nonnull Duration timeout, @Nonnull Duration pollPeriod)
+            throws NetworkException, TimeoutException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.runtime.exceptions.NetworkException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)